overlord/ifacestate/helpers.go

```diff
@@ -138,32 +138,55 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles(tm timings.Measurer) er
 	shouldWriteSystemKey := true
 	os.Remove(dirs.SnapSystemKeyFile)
 
-	// For each snap:
-	for _, snapInfo := range snaps {
-		snapName := snapInfo.InstanceName()
-		// Get the state of the snap so we can compute the confinement option
+	confinementOpts := func(snapName string) interfaces.ConfinementOptions {
 		var snapst snapstate.SnapState
 		if err := snapstate.Get(m.state, snapName, &snapst); err != nil {
 			logger.Noticef("cannot get state of snap %q: %s", snapName, err)
 		}
+		return confinementOptions(snapst.Flags)
+	}
 
-		// Compute confinement options
-		opts := confinementOptions(snapst.Flags)
+	// For each backend:
+	for _, backend := range securityBackends {
+		if backend.Name() == "" {
+			continue // Test backends have no name, skip them to simplify testing.
+		}
 
-		// For each backend:
-		for _, backend := range securityBackends {
-			if backend.Name() == "" {
-				continue // Test backends have no name, skip them to simplify testing.
-			}
-			// Refresh security of this snap and backend
-			timings.Run(tm, "setup-security-backend", fmt.Sprintf("setup security backend %q for snap %q", backend.Name(), snapInfo.InstanceName()), func(nesttm timings.Measurer) {
-				if err := backend.Setup(snapInfo, opts, m.repo, nesttm); err != nil {
+		// use .SetupMany() if implemented by the backend, otherwise fall back to .Setup()
+		if setupManyInterface, ok := backend.(interfaces.SecurityBackendSetupMany); ok {
```

Comment:
It might make sense to make this a problem of interfaces pkg though.  Have an interfaces.SetupMany helper there that calls SetupMany or does a loops depending on the interface impl.

Reply:
Done.